### PR TITLE
fix/site/toast-gtag-responsive

### PR DESCRIPTION
**Pull request by tianenpang**
- [x] fix copied toast
- [x] gtag script warning (next/script)
- [x] responsive mian
- [x] update range when current breakpoint changes
- [ ] disable button text selection

### CODE REVIEW

File: pages/components/[slug].tsx [hunk 1]
```diff
@@ -52,7 +52,7 @@ const Component: NextPage<Props> = ({ source, name, frontMatter }) => {
 
       <section>
         <div className="max-w-screen-xl px-4 pt-24 pb-16 mx-auto">
-          <div className="prose">
+          <div className="prose max-w-full">
```
Comment (markmead) [inline pages/components/[slug].tsx:55]: Both work but `max-w-none` is the suggested approach, can we use that, please?

File: styles/button.module.css [hunk 2]
```diff
@@ -1,3 +1,3 @@
 .pill {
-  @apply inline-flex items-center px-3 py-1.5 border-2 border-black rounded-lg hover:bg-black hover:text-white;
+  @apply inline-flex items-center px-3 py-1.5 border-2 border-black rounded-lg select-none hover:bg-black hover:text-white;
```
Comment (markmead) [inline styles/button.module.css:2]: Doesn't need this.

File: components/component/example.tsx [hunk 3]
```diff
@@ -112,9 +119,11 @@ const Test: FunctionComponent<Props> = ({ name, item, spacing }) => {
         </div>
 
         <div className={view ? 'hidden' : 'block'}>
-          <pre className="p-4 overflow-auto h-[400px] lg:h-[600px] ring-2 ring-black rounded-lg">
-            <code className="language-html">{code}</code>
-          </pre>
+          {!view && (
```
Comment (markmead) [inline components/component/example.tsx:122]: What's the benefit of this? 🤔

It's handled with the `className` above, feels like an extra step.

File: components/component/example.tsx [hunk 4]
```diff
@@ -66,6 +66,13 @@ const Test: FunctionComponent<Props> = ({ name, item, spacing }) => {
     range === 1348 ? setWidth('100%') : setWidth(`${range}px`)
   }, [range])
 
+  const handleWidth = (width: string) => {
+    setWidth(width)
+    width === '100%'
+      ? setRange(1348)
+      : setRange(Number(width.replace(/px$/, '')))
+  }
```
Comment (markmead) [inline components/component/example.tsx:74]: Great addition! Thank you! 🙌 

Could it be reformatted like: 

```js
const handleWidth = (width: string) => {
  setWidth(width)

  width === '100%'
    ? setRange(1348)
    : setRange(Number(width.replace('px', '')))
}
```

**Review by markmead — CHANGES_REQUESTED** [review]
Looks good! 🎉 There are a few comments, nothing huge!

---

I haven't had much look with `next/script` for Google Tag Manager, but I'll check out how it's doing on the staging link Vercel provided.

There were toast notifications, but I must have left them out in a recent update, here's the [commit where they were last working](https://github.com/markmead/hyperui/tree/464b9670faeb7aa0d4bba79e39a1cc3a6a70cdb8). It would be good to use what was there before, I don't mind making the change if you're OK with me making changes to this PR?

**Comment by tianenpang** [comment]
All done 🙌🏻

**Comment by markmead** [comment]
Awesome, thanks @tianenpang! Are you happy for me to make some changes to bring back in the old toast notifications?

**Comment by tianenpang** [comment]
Sure, please. I did not find the old toast in the [commit](https://github.com/markmead/hyperui/tree/464b9670faeb7aa0d4bba79e39a1cc3a6a70cdb8) you mentioned 👀

**Review by markmead — APPROVED** [review]
All good! Let's hope that `next/script` change doesn't break the tracking like it did the last time I tried it 😭

Feel free to merge when you see this @tianenpang!

**Comment by tianenpang** [comment]
I removed the toast style from the globals and the google analytics looks well tracked locally 👀

**Review by markmead — APPROVED** [review]

**Comment by tianenpang** [comment]
Hey @markmead, I've made some updates for google analytics feature, looks well collected in debug view and works fine in the preview. 👀

Screenshot of analytics debug view

Screenshot of debug view

**Comment by tianenpang** [comment]
Looks tracked very well and reverted to the way it was before.

Screenshot of analytics debug view

analytics debug view

**Comment by markmead** [comment]
Awesome! 🥳  Thanks for all the work on this, really appreciated!

**Pull request merged**